### ISSUE TITLE
Lazily load plugins?

### ISSUE THREAD

**Issue by maxfischer2781** [MatterMiners/cobald#119]
TARDIS has problems with shipping optional plugins because COBalD eagerly loads all plugins; this requires the entire plugin module to be loadable, *including all direct dependencies*. Similarly, all other global initialisation will be run unconditionally.

It seems preferable that COBalD would/could load plugins lazily.

**Comment by maxfischer2781** [comment]
The logic for loading, sorting and applying Plugins is spread over several places. This might be a good opportunity to streamline these things.  
TLDR: I think this is straightforward enough, **but** includes a breaking change of the API.

-----------

The Plugin metadata is held by a class

https://github.com/MatterMiners/cobald/blob/0343272cda484534132d20e2468d51f3f331f4e1/src/cobald/daemon/config/mapping.py#L113

which already supports creating it from a raw EntryPoint (the `section = module:qualname` of a distribution package)

https://github.com/MatterMiners/cobald/blob/0343272cda484534132d20e2468d51f3f331f4e1/src/cobald/daemon/config/mapping.py#L143-L144

at which point we could lazily load the requirements. The entrypoint already exposes the `section` name, so even without the requirements we can check whether a plugin is even needed.

The innermost function that needs the Plugins is 

https://github.com/MatterMiners/cobald/blob/0343272cda484534132d20e2468d51f3f331f4e1/src/cobald/daemon/config/mapping.py#L177

which also knows about the raw configuration. This could simply ignore optional Plugins for order resolution.

--------------

The complication is that Plugin requirements 

https://github.com/MatterMiners/cobald/blob/0343272cda484534132d20e2468d51f3f331f4e1/src/cobald/daemon/plugins.py#L34-L35

contain ordering information that we can ignore for sections that do not apply, and an absence-requirement check which we obviously cannot ignore for absent sections. If we want lazy Plugins, the "`require`" flag **must** be handled differently; this **will** break the existing API.

A simple approach would be to use a separate entry point key for regular versus required plugins. Since as far as I can tell (and hope), most plugins are **not** required I would propose `cobald.config.sections` and `cobald.config.required_sections`, respectively.
